src/rpc/blockchain.cpp

```diff
@@ -2021,13 +2021,15 @@ class CoinsViewScanReserver
         if (g_scan_in_progress.exchange(true)) {
             return false;
         }
+        CHECK_NONFATAL(g_scan_progress == 0);
```

Comment:
If this were to fail, wouldn't it likely block all future reserves?

Reply:
You mean this should be fatal?

Reply:
Since it should be impossible, maybe simplest solution is to just remove it.

Otherwise, it might need something like:
```suggestion
        if (g_scan_progress) {
            error("ERROR: g_scan_progress was %s when it should be 0, in %s", g_scan_progress, __func__);
            g_scan_progress = 0;
        }
```

Or:
```suggestion
        if (g_scan_progress) {
            g_scan_in_progress = false;
            CHECK_NONFATAL(g_scan_progress == 0);
        }
```

Reply:
Then this should be `assert(g_scan_progress == 0)`. I don't think all RPC code should use `CHECK_NONFATAL`, at least #17192 doesn't say it.

Reply:
You can restart your node if this fails, but I don't think there is a need to crash the node.